### PR TITLE
Add docs to `otel_ctx` and `otel_baggage`

### PR DESCRIPTION
**Pull request by whatyouhide**
Moar docs 🙃

### CODE REVIEW

File: apps/opentelemetry_api/src/otel_ctx.erl [hunk 1]
```diff
@@ -41,11 +45,17 @@
          text_map_injector_fun/3]).
 
 -type t() :: map() | undefined.
+%% A context map.
+
 -type key() :: term().
+%% A context key.
```
Comment (tsloughter) [inline apps/opentelemetry_api/src/otel_ctx.erl:51]: For consistency with the rest of the project the comment about the type should go above the type, not below it.

**Comment by tsloughter** [comment]
Looks good except for my comment about the comments going above the definitions.

**Comment by tsloughter** [comment]
This reminded me we really need macros for Ctx and Baggage. I dislike that we have any functions that work on the pdict instead of explicit context :(. I think it'd be clearer if it was only macros that ever read the context from the pdict. Then those functions would be deprecated.

Of course, not something that has to be addressed in this PR. Though I am curious about your thoughts on that.

File: apps/opentelemetry_api/src/otel_ctx.erl (hunk 1 above)
Comment (whatyouhide) [reply apps/opentelemetry_api/src/otel_ctx.erl:51]: @tsloughter "above the type" doesn't show up in edoc, that's why I went with below the type.

This is with the comment above the type:

CleanShot 2024-02-05 at 12 11 49@2x

Comment (tsloughter) [reply apps/opentelemetry_api/src/otel_ctx.erl:51]: Argh oops, ok, haha, nevermind then :)

**Review by tsloughter — APPROVED** [review]

**Comment by whatyouhide** [comment]
@tsloughter if we're only ever meant to get the context from the process dictionary then yes, would be nice to deprecate the "explicit context" functions. I’m pretty much always for smaller API surfaces 😄

**Comment by tsloughter** [comment]
Oh na, other way around. Deprecate the ones that do implicit context and only do that in macros that then call the functions with the context.

**Comment by whatyouhide** [comment]
@tsloughter ah, even better then yes. 🙃

**Pull request merged**